components/scream/src/physics/p3/tests/p3_cloud_rain_acc_unit_tests.cpp

```diff
@@ -0,0 +1,101 @@
+#include "catch2/catch.hpp"
+
+#include "share/scream_types.hpp"
+#include "share/util/scream_utils.hpp"
+#include "share/scream_kokkos.hpp"
+#include "share/scream_pack.hpp"
+#include "physics/p3/p3_functions.hpp"
+#include "physics/p3/p3_functions_f90.hpp"
+#include "share/util/scream_kokkos_utils.hpp"
+
+#include "p3_unit_tests_common.hpp"
+
+#include <thread>
+#include <array>
+#include <algorithm>
+#include <random>
+
+namespace scream {
+namespace p3 {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestCloudRainAccretion {
+
+static void run_phys()
+{
+  // TODO
+}
+
+static void run_bfb()
+{
+  // This is the threshold for whether the qc and qr cloud mixing ratios are
+  // large enough to affect the warm-phase process rates qcacc and ncacc.
+  constexpr Scalar qsmall = C::QSMALL;
+
+  constexpr Scalar rho = 1.0;
+  constexpr Scalar inv_rho = 1.0 / rho;
+  constexpr Scalar qc_incld_small = 0.9 * qsmall;
+  constexpr Scalar qr_incld_small = 0.9 * qsmall;
+  constexpr Scalar qc_incld_not_small = 2.0 * qsmall;
+  constexpr Scalar qr_incld_not_small = 2.0 * qsmall;
+  constexpr Scalar nc_incld = 1.0;
+
+  CloudRainAccretionData fortran_data[] = {
+      // rho, inv_rho, qc_incld, nc_incld, qr_incld, qcacc, ncacc
+      {rho, inv_rho, qc_incld_small, nc_incld, qr_incld_small, 0.0, 0.0},
+      {rho, inv_rho, qc_incld_small, nc_incld, qr_incld_not_small, 0.0, 0.0},
+      {rho, inv_rho, qc_incld_not_small, nc_incld, qr_incld_small, 0.0, 0.0},
+      {rho, inv_rho, qc_incld_not_small, nc_incld, qr_incld_not_small, 0.0, 0.0},
+  };
+
+  static constexpr Int num_runs = sizeof(fortran_data) / sizeof(CloudRainAccretionData);
+
+  // Create copies of data for use by cxx. Needs to happen before fortran calls so that
+  // inout data is in original state
+  CloudRainAccretionData cxx_data[] = {
+      fortran_data[0],
+      fortran_data[1],
+      fortran_data[2],
+      fortran_data[3]
+  };
+
+  // Run the Fortran subroutine.
+  for (Int i = 0; i < num_runs; ++i) {
+    cloud_rain_accretion(fortran_data[i]);
+  }
+
+  // Run the C++ subroutine.
+  for (Int i = 0; i < num_runs; ++i) {
+    CloudRainAccretionData& d = cxx_data[i];
```

Comment:
Unfortunately, this pattern won't work. The _f functions convert the incoming scalars into uniform packs as a hack. For the unit tests, we want to test non-uniform packs. See the pattern in the bfb unit tests in p3_ice_collection_unit_tests.cpp

Reply:
Cool, thanks for the pointer.

Reply:
Should I emulate the pack size logic in the ice collection unit test as well? I can pull in the device sync stuff too. I copied this from another unit test that didn't have those things, so I can use this test as a model for the others.

Reply:
Yes, that test follows the correct model for unit-tests of functions that just take single packs of data.